ambry-store/src/main/java/com/github/ambry/store/BlobStore.java

```diff
@@ -772,6 +775,183 @@ public void delete(List<MessageInfo> infosToDelete) throws StoreException {
     }
   }
 
+  @Override
+  public StoreBatchDeleteInfo batchDelete(List<MessageInfo> infosToDelete) throws StoreException {
+    checkStarted();
+    checkDuplicates(infosToDelete);
+    StoreBatchDeleteInfo storeBatchDeleteInfo = new StoreBatchDeleteInfo(
+        replicaId.getPartitionId(), new ArrayList<>());
+    final Timer.Context context = metrics.batchDeleteResponse.time();
+    int i = 0;
+    List<MessageInfo> filteredInfos = null;
+    try {
+      List<IndexValue> indexValuesPriorToDelete = new ArrayList<>();
+      List<IndexValue> originalPuts = new ArrayList<>();
+      List<Short> lifeVersions = new ArrayList<>();
+      Offset indexEndOffsetBeforeCheck = index.getCurrentEndOffset();
+      filteredInfos = new ArrayList<>();
+      for (MessageInfo info : infosToDelete) {
+        try {
+          IndexValue value =
+              index.findKey(info.getStoreKey(), new FileSpan(index.getStartOffset(), indexEndOffsetBeforeCheck));
+          if (value == null) {
+            throw new StoreException(
```

Comment:
Seems the infoToDelete are iterated twice

    validate and filter only the qualified Ids
    Issue delete for the ids identified via#1

Iteration(1) seems to be throwing error when an id is not present in index..etc, think we need to handle this exception local to the function and ensure the batchDelete operation in attempted in fullest.